pkg/controller/toolchainconfig/toolchainconfig_controller_test.go

```diff
@@ -0,0 +1,105 @@
+package toolchainconfig
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestReconcileWhenToolchainConfigIsAvailable(t *testing.T) {
+	// given
+	config := newToolchainConfigWithReset(t, AutomaticApprovalCfg().MaxUsersNumber(123, PerMemberClusterCfg("member1", 321)))
+	cl := NewFakeClient(t, config)
+	controller := Reconciler{
+		Client: cl,
+		Log:    ctrl.Log.WithName("controllers").WithName("ToolchainConfig"),
+	}
+
+	// when
+	_, err := controller.Reconcile(newRequest())
+
+	// then
+	require.NoError(t, err)
+	actual, err := GetConfig(NewFakeClient(t), HostOperatorNs)
+	require.NoError(t, err)
+	assert.False(t, actual.AutomaticApproval().IsEnabled())
+	assert.Equal(t, 123, actual.AutomaticApproval().MaxNumberOfUsersOverall())
```

Comment:
Same here. Don't really need to test all these params.

Reply:
Done in https://github.com/codeready-toolchain/host-operator/pull/438/commits/e5188e1c6702fa0801f91dea0520ec7e3a0da5de